hrpc/compact.go

```diff
@@ -0,0 +1,57 @@
+package hrpc
+
+import (
+	"context"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/tsuna/gohbase/pb"
+)
+
+type CompactRegion struct {
+	tableOp
+
+	region string
+	major  bool
+	family []byte
+}
+
+//Create a new CompactRegion struct
+func NewCompactRegion(region string, family []byte, major bool) *CompactRegion {
+	chunks := strings.Split(region, ",")
+	return &CompactRegion{
+		tableOp: tableOp{
+			base: base{
+				ctx:   context.Background(),
+				table: []byte(chunks[0]),
+				key:   []byte(chunks[1]),
+			},
+		},
+		region: region,
+		family: family,
+		major:  major,
+	}
+}
+
+//Return the name of this RPC  method
+func (cr *CompactRegion) Name() string {
+	return "CompactRegion"
```

Comment:
my suspicion is this "CompactRegion" is a wrong name for the rpc, because that's one of the places where `UnsupportedOperationException` is thrown: https://github.com/apache/hbase/blob/branch-1.3/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java#L1974